File: FixFlow/FixFlow.Application/DTOs/Request/UpdateJobStatusRequest.cs
using System.ComponentModel.DataAnnotations;

namespace FixFlow.Application.DTOs.Request;

public class UpdateJobStatusRequest
{
    [Required(ErrorMessage = "Novi status je obavezan.")]
    public int NewStatus { get; set; }

    [StringLength(500, ErrorMessage = "Napomena ne može biti duža od 500 karaktera.")]
    public string? Note { get; set; }
}

File: packages/fixflow_core/lib/models/requests/update_job_status_request.dart
class UpdateJobStatusRequest {
  final int newStatus;
  final String? note;

  UpdateJobStatusRequest({
    required this.newStatus,
    this.note,
  });

  Map<String, dynamic> toJson() => {
        'newStatus': newStatus,
        if (note != null) 'note': note,
      };
}
